packages/amplify_core/lib/src/types/api/async/legacy_api_operation.dart

```diff
@@ -13,11 +13,12 @@
  * permissions and limitations under the License.
  */
 
-import 'package:amplify_core/amplify_core.dart';
+import 'package:async/async.dart';
 
-class GraphQLOperation<T> {
-  final Function cancel;
-  final Future<GraphQLResponse<T>> response;
-
-  const GraphQLOperation({required this.response, required this.cancel});
+/// Eventually this should be deprecated and just use [CancelableOperation].
+/// Until then, this is used to make `.response` available like it was for older
+/// [GraphQLOperation] and [RestOperation] classes.
+extension LegacyApiOperation<T> on CancelableOperation<T> {
+  @Deprecated('Use .value instead.')
```

Comment:
Based on the updates in the example app, it seems like customers will (probably) still need to make changes even after changing `.response` to `.value` since value/response is now a different type than value use to be, correct? If so, maybe the auto migration is less useful. Still something to consider. I'll let you determine if it is worth it or not.

Since the type is changes, I'm actually not sure how useful it is to have `.response` at all. This is similar to a comment a left farther down, but to me the deprecated annotation should be used when something *will( break, not when it has already been broken. Since (based on the examples) it looks like codes changes are required, I would consider this already broken.

I guess it could be helpful to have the message. The message probably needs more info though. Again, will leave it up to you to decide how much value this adds to the customer.

Reply:
spoke offline about this

Reply:
While yes, the operation types are changing in breaking way for someone who refers to those types, the .response => .value proxy is there because I suspect there are people  who don't refer to the types, just to the .response field, so this makes it easier for that subset of people. Agree it is breaking already and not textbook use of deprecation annotation.